### PR TITLE
Slideshow Shortcode: Adding the Jetpack namespace to CSS class names

### LINKED ISSUE #17715: Slideshow shortcode: stylesheet doesn't have namespaced classes
#### Steps to reproduce the issue

[Slideshow class names](https://github.com/Automattic/jetpack/blob/master/projects/plugins/jetpack/modules/shortcodes/css/slideshow-shortcode.css) seem too common to me and created an issue on a project I'm working on where a carousel is picking up styling from the above Jetpack file that it shouldn't.

#### What I expected

I expect the styling classes in Jetpack to be namespaced and not conflict with common class naming.

#### What happened instead

The carousel stylesheet ([slideshow-shortcode.css](https://github.com/Automattic/jetpack/blob/master/modules/shortcodes/css/slideshow-shortcode.css)) don't have "jetpack" prefixed to all the classes whereas the other files in [the same directory](https://github.com/Automattic/jetpack/tree/master/modules/shortcodes/css) do. For example, in the recipes.css file classes look like `.jetpack-recipe` and in in the quiz.css file they look like `div.jetpack-quiz` but in the slideshow files they look like `.slideshow-loading`.

### PR DESCRIPTION
**Pull request by coder-karen**
Fixes #17715

#### Changes proposed in this Pull Request:

* Changes proposed in this PR include changes to CSS classes within the [slideshow-shortcode.css ](https://github.com/Automattic/jetpack/blob/master/projects/plugins/jetpack/modules/shortcodes/css/slideshow-shortcode.css) file (prefixing them with `jetpack-`). 
* This is to prevent CSS name collisions. 
* This involves changes to the CSS file as well as the PHP and JS files where the classnames are first defined. It also involves changes to theme compatibility files which reference some of the CSS classes (for [TwentyFourteen](https://github.com/Automattic/jetpack/blob/master/projects/plugins/jetpack/modules/theme-tools/compat/twentyfourteen.css), [TwentyFifteen](https://github.com/Automattic/jetpack/blob/master/projects/plugins/jetpack/modules/theme-tools/compat/twentyfifteen.css) and [TwentySixteen](https://github.com/Automattic/jetpack/blob/master/projects/plugins/jetpack/modules/theme-tools/compat/twentysixteen.css)). Lastly, it also includes a change to the [test-class-slideshow](https://github.com/Automattic/jetpack/blob/master/projects/plugins/jetpack/tests/php/modules/shortcodes/test-class.slideshow.php) PHP file to include an updated CSS classname where relevant.

#### Additional comments:

* In testing the theme compat files, I noticed no change on my test site to classnames (when viewing in the inspector) either before or after changing the CSS classnames in those files. That seemed to be because all slideshows were already using the `jetpack` namespaced classnames after the change to the three main slideshow-shortcode files (css, js and php). It may be that there is another way to test slideshows that doesn’t use block classnames, however the theme compat files have been changed anyway to be safe.

#### Jetpack product discussion:

* Internal discussion here: p1643907890247239-slack-CBG1CP4EN

#### Does this pull request change what data or activity we track or use?

n/a

#### Testing instructions:

* To test, add a slideshow to a post or page using the shortcode. Here is an example:  `[gallery type="slideshow" ids="22,21,19,20"]` - however you’d need to replace those IDs with image IDs from your test site. The end result should look similar to this:

Slideshow-via-shortcode

* Other ways to test the shortcode version of the slideshow include adding the Classic Widgets plugin, and adding a gallery widget to a test site, making sure the gallery is a slideshow. Checking the inspector should confirm that the jetpack prefix is being added to class names (examples will include `jetpack-slideshow-window`, `jetpack-slideshow-slide`).

### CODE REVIEW

**Comment by coder-karen** [comment]
Tested the changes on a WordPress.com simple site, using the shortcode block to add the slideshow shortcode (and also tested using the TwentySixteen theme), the CSss is picked up and the slideshow looks and acts as it should, with no difference.

**Review by samiff — COMMENTED** [review]
Looks good to me, nice job 👍 

I did notice that the wpcom diff is lacking the twentyfourteen and twentyfifteen changes. @jeherve should we be Fusioning those theme compat files as well? At quick glance the wpcom copies look mostly in sync and untouched for some time.

**Comment by kraftbj** [comment]
Something worth doing since this is changing the CSS selector, we should review these selectors in https://wpdirectory.net to see if there any themes styling against our slideshows that may be impacted (or to put a different way, shows us the themes we might be breaking right now that this would fix :) ).

**Comment by coder-karen** [comment]
> we should review these selectors in [wpdirectory.net](https://wpdirectory.net/) to see if there any themes styling against our slideshows that may be impacted (or to put a different way, shows us the themes we might be breaking right now that this would fix :) ).

Oh yes, thank you for mentioning this @kraftbj ! Deep dive below:

**Checking [slideshow-slide](https://wpdirectory.net/search/01FVB5H63MGTDQNWR3V5H6C0ZE):**

Of 34 themes with active installations, 15 refer to different class-names so aren’t affected by changes.

Of the remaining 19 themes, quite a few only have minimal styling changes that are affected with the new class name introduction.

A few stand out though:
- Those that have significant styling changes (where the Jetpack namespaces classes more significantly break the styling added by the theme) include Cubic, Boardwalk, Flat Responsive, Jason Lite, Mighty and Styled Mag. Of those, Cubic had the most active installs with 3000+. Cubic and Boardwalk (800+ active installs) were updated the longest time ago (June 2017).
- Goran was the only theme that benefited in one respect from the styling changes (caption text was now readable in my testing)
- Activating Maskitto Light resulted in my test site crashing (with the updated Jetpack-namespaces classnames already added). It activated fine on a test site without the changes. However, that theme hasn’t been updated since June 2016 (with 1000+ active installs)

None of the themes had shown any significant shortcode slideshow conflicts with Jetpack prior to the CSS name change.

**Checking [slideshow-window](https://wpdirectory.net/search/01FVCFG0ZKV9MSFMYT6403CM58):**

There were more additional themes not mentioned in the above search (42). Of those, there were none with overly significant differences in styling - at the most some border cooler changes, border removal or border radius changes). Three had slightly more obvious (but non breaking) changes: Jefferson, Bayn Lite and Magnetism.

**Additionally, checking plugins:**

- The [.slideshow-window](https://wpdirectory.net/search/01FTZW6JJZ5ZAGAS11FPE9CDJ4) class is referenced by several plugins that extend Jetpack, however the most up to date of those was last updated two years ago.
- The [.slideshow-slide](https://wpdirectory.net/search/01FVCJAJY0D98NYZRCJRNBSPHE) class is referenced by the same plugins as above with one exception - the addition of the Hide My WP Ghost plugin (last update 2 weeks ago, 100,000+ active installs). Similar to that is the Hide Wp-Admin and Wp-Login plugin (900+ active installs) and Website Security Check (100+ active installs). There are a few other plugins mentioned but they reference slightly different classnames (not Jetpack classes).

---

In short, changing these CSS class names will affect multiple themes and plugins. In one theme this may result in a breaking change if Jetpack is also installed and active, however in all other cases changes are mostly small. We also no longer have information in our support docs about how to add slideshows via shortcode with Jetpack, meaning usage is likely to reduce over time. I'm now on the fence as to whether the pros outweigh the cons so would like some additional opinions on this.

**Comment by coder-karen** [comment]
Noting here that I've updated the above comment with an edit as the one theme resulting in a crashed site was not related to Jetpack.

**Comment by samiff** [comment]
On one hand, I'd prefer that Jetpack does things "the right way" by not using general class names like this PR addresses. On the other hand, I'd like to avoid creating issues with existing content on sites if possible.

Given that the slideshow shortcode is an older feature, and that newer content/sites could be using the arguably better Slideshow block (among other options), I wonder if "fixing" this would be a net negative. @kraftbj @jeherve Where do you land on this?

**Comment by kraftbj** [comment]
I think this is a good move to make—if nothing else get the styling in stable place to put into a long-term maintenance mode (as blocks will be the way). Cubic and Boardwalk are Automattic themes; we can delay this to inquire if the Themes team want to update them or not.

**Comment by coder-karen** [comment]
> Cubic and Boardwalk are Automattic themes; we can delay this to inquire if the Themes team want to update them or not.

Sounds good. I've reached out: p1644402685939439-slack-C029FM1EH

**Comment by samiff** [comment]
@coder-karen I've set this to In Progress while you work with Themes to coordinate any updates 👍

**Comment by coder-karen** [comment]
Adding a comment here to say that the goal is to add this to the 10.8 release.

**Comment by coder-karen** [comment]
Opening this up again for review as the Themes team are also ready for this change now.

**Review by samiff — APPROVED** [review]
I've fixed up the Fusion in D77671-code

So this PR's diff should be ready to go now: D74445-code

@coder-karen Can I land D74445 now, or do we need to coordinate with any changes Themes team is doing?

**Comment by coder-karen** [comment]
Themes team is ready - p1648210978716059/1644402685.939439-slack-C029FM1EH - so D74445 can be landed, thanks @samiff !

**Comment by samiff** [comment]
D74445-code deployed in r242729-wpcom
